objectstore.go

```diff
@@ -117,13 +133,45 @@ func (o *ObjectStore) ListObjects(bucket, prefix string) ([]string, error) {
 
 func (o *ObjectStore) DeleteObject(bucket, key string) error {
 	o.log.Infof("objectStore.DeleteObject called")
-	if _, err := o.project.DeleteObject(context.Background(), bucket, key); err != nil {
+	project, err := uplink.OpenProject(context.Background(), o.access)
+	if err != nil {
+		return err
+	}
+	defer project.Close()
+
+	if _, err := project.DeleteObject(context.Background(), bucket, key); err != nil {
 		return err
 	}
 	return nil
 }
 
 func (o *ObjectStore) CreateSignedURL(bucket, key string, ttl time.Duration) (string, error) {
 	o.log.Infof("objectStore.CreateSignedURL called")
-	return "", errors.New("CreateSignedURL is not supported for this plugin")
+	var sharePrefixes []uplink.SharePrefix
+	sharePrefixes = append(sharePrefixes, uplink.SharePrefix{
+		Bucket: bucket,
+		Prefix: key,
+	})
+
+	permission := uplink.Permission{}
+	permission.AllowDownload = true
+	permission.AllowDelete = false
+	permission.AllowUpload = false
+	permission.AllowList = true // ?
```

Comment:
You can replace the above 5 lines with `permission := uplink.ReadOnlyPermission()`